feat: add `assert/has-float16array-support`

udaykakade25 opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   Adds assert/has-float16array-support

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   adds support for detecting `Float16Array` support.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

This is the second PR (apart from the first PR: assert/is-float16array) that adds support for float16 arrays, requiring the creation of an array/float16. The first PR ( #7273 ) should be merged first, as this PR requires some functionality from the previous PR.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

udaykakade25 committed:
add assert/has-float16array-support

udaykakade25 committed:
Removed extra line at the end of code in index.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 commented:
Hello @Planeshifter and @kgryte sir,
This is the second PR that requires the first PR (#7273) to be merged, as some functionalities are taken from it.
Almost every code file is running perfectly after skipping test cases requiring float16.
Please do review this PR!

kgryte commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into hasFloat16ArraySupport

kgryte commented on lib/node_modules/@stdlib/assert/has-float16array-support/README.md:
````diff
@@ -0,0 +1,123 @@
+### Usage
+
+```text
+Usage: has-float32array-support [options]
````
```suggestion
Usage: has-float16array-support [options]
```

kgryte committed:
docs: fix command name

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/assert/has-float16array-support/lib/main.js:
```diff
@@ -0,0 +1,65 @@
+// MODULES //
+
+var isFloat16Array = require( '@stdlib/assert/is-float16array' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
```
```suggestion
var PINF = require( '@stdlib/constants/float64/pinf' ); // TODO: replace with `constants/float16/pinf`
```

kgryte committed:
docs: add TODO

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/assert/has-float16array-support/lib/float16array.js:
```diff
@@ -0,0 +1,28 @@
+
+// MAIN //
+
+var main = ( typeof Float16Array === 'function' ) ? Float16Array : null;
```
```suggestion
var main = ( typeof Float16Array === 'function' ) ? Float16Array : null; // eslint-disable-line stdlib/require-globals
```

kgryte committed:
docs: disable lint rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/assert/has-float16array-support/lib/main.js:
```diff
@@ -0,0 +1,65 @@
+	}
+	// Test basic support...
+	try {
+		arr = new GlobalFloat16Array( [ 1.0, 3.14, -3.14, 5.0e40 ] );
+		bool = (
+			isFloat16Array( arr ) &&
+			arr[ 0 ] === 1.0 &&
+			arr[ 1 ] !== 3.14 &&
+			arr[ 2 ] !== -3.14 &&
+			arr[ 3 ] === PINF
```
```suggestion
		arr = new GlobalFloat16Array( [ 1.0, 3.0, -3.0, 5.0e40 ] );
		bool = (
			isFloat16Array( arr ) &&
			arr[ 0 ] === 1.0 &&
			arr[ 1 ] !== 3.0 &&
			arr[ 2 ] !== -3.0 &&
			arr[ 3 ] === PINF
```

kgryte committed:
fix: avoid rounding errors

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/assert/has-float16array-support/package.json:
```diff
@@ -0,0 +1,74 @@
+    "detect",
+    "feature",
+    "float16array",
+    "es2022",
```
```suggestion
    "es2025",
```

kgryte committed:
chore: fix keyword

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into hasFloat16ArraySupport

kgryte committed:
chore: resolve style issues and clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
test: fix test case

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (dismissed):
LGTM after clean-up.

kgryte committed:
style: disable lint rule

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
fix: resolve failing tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---
